packages/flutter/lib/src/widgets/two_dimensional_viewport.dart

```diff
@@ -921,9 +917,298 @@ abstract class RenderTwoDimensionalViewport extends RenderBox implements RenderA
   }
 
   @override
-  RevealedOffset getOffsetToReveal(RenderObject target, double alignment, { Rect? rect }) {
-    // TODO(Piinks): Add this back in follow up change (ensureVisible), https://github.com/flutter/flutter/issues/126299
-    return const RevealedOffset(offset: 0.0, rect: Rect.zero);
+  RevealedOffset getOffsetToReveal(
+    RenderObject target,
+    double alignment, {
+    Rect? rect,
+    AxisDirection? axisDirection,
+  }) {
+    // We must know which axis we are revealing for, since Offset refers to only
+    // one of two scroll positions.
+    assert(axisDirection != null);
+
+    // Starting at `target` and walking towards the root:
+    //  - `child` will be the last object before we reach this viewport, and
+    //  - `pivot` will be the last RenderBox before we reach this viewport.
+    RenderObject child = target;
+    RenderBox? pivot;
+    // Find the direct child of the viewport that contains our target
+    while (child.parent != this) {
+      final RenderObject parent = child.parent!;
+      if (child is RenderBox) {
+        pivot = child;
+      }
+      child = parent;
```

Comment:
It looks like here we assume `target ... this` are all render boxes? In that case I think we can just compute the local `rect` using `getTransform(target)` and get the desired paint offset directly without having to do the while loop here?

Reply:
Also on a different note it looks like `getOffsetToReveal` should be `@protected` since it doesn't work on any `target` that's a descendant.

Reply:
Fixed!